src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -1759,11 +1880,18 @@ bool isRefEscape
             ImmutableArray<RefKind> argRefKindsOpt,
             ImmutableArray<int> argsToParamsOpt,
             bool ignoreArglistRefKinds,
```

Comment:
We should rely on the `this` parameter to provide the RefKind to decide whether references can escape into the receiver